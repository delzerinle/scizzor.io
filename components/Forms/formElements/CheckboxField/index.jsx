import PropTypes from 'prop-types';
import { Field } from 'formik';

const CheckboxField = ({ name, label, errors, touched, value, className }) => (
  <div className={className}>
    <Field name={name}>
      {({ field, form }) => (
        <input
          id={value}
          {...field}
          type="checkbox"
          checked={field.value === value}
          onChange={() => {
            field.onChange(field.name)(value);
            form.setFieldTouched(field.name, true);
          }}
          className="form-checkbox text-primary h-5 w-5 mr-3"
        />
      )}
    </Field>
    <label className="font-light text-sm" htmlFor={value}>
      {label}
    </label>
  </div>
);

CheckboxField.propTypes = {
  errors: PropTypes.object,
  touched: PropTypes.object,
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default CheckboxField;
