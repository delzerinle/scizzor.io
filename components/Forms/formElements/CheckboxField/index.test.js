import React from 'react';
import { shallow } from 'enzyme';

import CheckboxField from './index';

describe('Checkbox field tests', () => {
  it('renders without crashing', () => {
    shallow(
      <CheckboxField name="test" label="test" value="test" className="test" />
    );
  });

  it('checkbox field snapshot', () => {
    const wrapper = shallow(
      <CheckboxField name="test" label="test" value="test" className="test" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
