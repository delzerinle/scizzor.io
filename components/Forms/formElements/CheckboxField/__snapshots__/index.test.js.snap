// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox field tests checkbox field snapshot 1`] = `
<div
  className="test"
>
  <Field
    name="test"
  >
    <Component />
  </Field>
  <label
    className="font-light text-sm"
    htmlFor="test"
  >
    test
  </label>
</div>
`;
